mtoc 927.0.2 (new formula)

roolebo opened the pull request:
mtoc is a Mach-O to PE binary converter.
It enables Tianocore UEFI build on macOS.

- [*] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [*] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [*] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [*] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [*] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
The formula is based off #19922 made by @rickmark. Changelog:
* mtoc is updated to the latest available version (949.0.1 can't be downloaded);
* install routine is updated to match the new version;
* man page is installed and test was added.

SMillerDev commented on Formula/mtoc.rb:
```diff
@@ -0,0 +1,39 @@
+  url "https://opensource.apple.com/tarballs/cctools/cctools-927.0.2.tar.gz"
+  sha256 "3d5d35e7120aac187de306019d4b53b3b08dd7ebf48764fcd5c8cefef40b3dbf"
+  depends_on "llvm" => :build
+
```
```suggestion
  sha256 "3d5d35e7120aac187de306019d4b53b3b08dd7ebf48764fcd5c8cefef40b3dbf"

  depends_on "llvm" => :build

```

roolebo replied:
done, thanks

SMillerDev commented on Formula/mtoc.rb:
```diff
@@ -0,0 +1,39 @@
+  def install
+    build_include = "#{buildpath}/include"
+    cp_r "#{HOMEBREW_PREFIX}/opt/llvm/include/llvm", build_include
+    cp_r "#{HOMEBREW_PREFIX}/opt/llvm/include/llvm-c", build_include
```
```suggestion
    cp_r "#{Formula['llvm'].include}/llvm", build_include
    cp_r "#{Formula['llvm'].include}/llvm-c", build_include
```

roolebo replied:
done, thanks

SMillerDev commented on Formula/mtoc.rb:
```diff
@@ -0,0 +1,39 @@
+    cp_r "#{HOMEBREW_PREFIX}/opt/llvm/include/llvm-c", build_include
+
+    # Conflicts with _structs.h in MacOSX10.15.sdk
+    File.delete "include/mach/i386/_structs.h" if MacOS.version == :catalina
```
This should probably be fixed upstream instead of being hotfixed here.

rickmark replied:
Seems like upstream may be internal Apple SDKs sadly....

roolebo replied:
I'd be glad to send a fix upstream but there's no way to do that as @rickmark noticed.

Bo98 replied:
It could well be fixed in 949.0.1 that is ["there"](https://opensource.apple.com/tarballs/cctools/) but the link is broken.

roolebo replied:
@Bo98, yeah I wasn't able to use the link as well. I mentioned that in the PR

roolebo commented:
@BrewTestBot is right but I don't know how to specify that the packag

roolebo commented:
I'm sorry I have no idea how to reopen PR, GitHub does let me to do that.

Bo98 commented:
Looking at the Jenkins logs, it looks like there is some _structs.h related compilation errors on Mojave and High Sierra as well.

roolebo commented:
@Bo98 Thanks, I have corrected it, the build does pass. Do I have to squash all commits in the PR?

Bo98 commented:
Ideally, but I can do it while merging if you prefer.

roolebo commented:
@Bo98 I squashed it and removed header copying. I also tried to use built-in llvm with ```uses_from_macos "llvm"``` instead of homebrew llvm but it fails with:
```
../lto.c:8:10: fatal error: 'llvm-c/lto.h' file not found
#include <llvm-c/lto.h>
         ^~~~~~~~~~~~~~
1 error generated.
make[1]: *** [lto.o] Error 1
```
Indeed macOS SDK doesn't include the file.
